config/management/genesis/src/fullnode_builder.rs

```diff
@@ -84,6 +109,58 @@ impl FullnodeConfig {
         }
     }
 
+    fn attach_to_validator(&mut self, validator_config: &mut NodeConfig) -> Result<()> {
+        ensure!(
+            matches!(validator_config.base.role, RoleType::Validator),
+            "Validator config must be a Validator config"
+        );
+
+        // Grab the public network config from the validator and insert it into the VFN's config
+        let public_network = {
+            let (i, _) = validator_config
+                .full_node_networks
+                .iter()
+                .enumerate()
+                .find(|(_i, config)| config.network_id == NetworkId::Public)
+                .expect("Validator should have a public network");
+            validator_config.full_node_networks.remove(i)
+        };
+
+        let fullnode_public_network = self
+            .config
+            .full_node_networks
+            .iter_mut()
+            .find(|config| config.network_id == NetworkId::Public)
+            .expect("VFN should have a public network");
+        fullnode_public_network.identity = public_network.identity;
+        fullnode_public_network.listen_address = public_network.listen_address;
+
+        // Grab the validator's vfn network information and configure it as a seed for the VFN's
+        // vfn network
+        let validators_vfn_config = validator_config
+            .full_node_networks
+            .iter()
+            .find(|config| config.network_id.is_vfn_network())
+            .expect("Validator should have vfn network");
+
+        let fullnode_vfn_network = self
```

Comment:
I'd name these the same to prevent confusion:

`validator_vfn_network` & `fullnode_vfn_network`